### PR TITLE
[SMALLFIX] [branch-1.6] Throw exception when underlying BlockInStream is inconsistent

### PR DESCRIPTION
**Pull request by apc999**
This fix the case when underlying BlockInStream is actually shorter than expected. E.g., the files are out of sync. In this case, this PR improves the client to throw an exception indicating the inconsistency. Otherwise, the client is returning meaningless data.

### CODE REVIEW

**Comment by apc999** [comment]
Jenkins, test this please

File: core/client/fs/src/main/java/alluxio/client/block/stream/BlockInStream.java [hunk 1]
```diff
@@ -201,6 +201,10 @@ public int read(byte[] b, int off, int len) throws IOException {
     }
     if (mEOF) {
       closePacketReader();
+      Preconditions.checkState(mPos >= mLength,
+          "Block %s is expected to be %s bytes, but only %s bytes are available. Please ensure the"
+              + "the file metadata is consistent between Alluxio and UFS.", mId, mLength,
+          mPos);
```
Comment (calvinjia) [inline core/client/fs/src/main/java/alluxio/client/block/stream/BlockInStream.java]: Make this a precondition message?

Comment (apc999) [reply core/client/fs/src/main/java/alluxio/client/block/stream/BlockInStream.java]: fixed

**Comment by apc999** [comment]
@calvinjia PTAL

**Pull request merged**